lib/checkstl.cpp

```diff
@@ -772,6 +781,83 @@ void CheckStl::mismatchingContainers()
     }
 }
 
+static bool isInvalidMethod(const Token * tok)
+{
+    if (Token::Match(tok->next(), ". assign|clear"))
```

Comment:
the goal is to move the patterns into the configuration files:
 * I have the feeling these checkers could be used to check boost/qt/wxwidgets/etc usage also.
 * I think it's ugly to hardcode the function names like this.

You did not invent this hardcoding but well it feels bad to continue on this path.

Reply:
I think you can ignore this comment for a moment. But please keep it in mind in the future. The `CheckStl` needs deep refactorings. If you take care of my other review comments I think this can be merged.

Reply:
> I have the feeling these checkers could be used to check boost/qt/wxwidgets/etc usage also.

Qt is the only outlier. All most all major libraries and frameworks(including wxwidgets and boost) use C++'s definition of containers.

We can update the configuration for this in the future to support Qt or possible extensions to C++ containers in libraries.